modules/default/calendar/calendar.js

```diff
@@ -93,6 +93,22 @@ Module.register("calendar", {
 		// indicate no data available yet
 		this.loaded = false;
 
+		if (this.config.timeFormat === "relative") {
+			this.fullDayEventRelativeCalendar = {};
+			let calendarTemplate = this.Translator.coreTranslations["FULL_DAY_EVENT_RELATIVE_CALENDAR"];
+			if (calendarTemplate) {
+				Object.entries(calendarTemplate).forEach(([key, value]) => {
+					if (value && value.startsWith("function")) {
+						value = "(" + value + ")";
+					}
+					if (value && value.startsWith("(function")) {
+						value = eval(value);
```

Comment:
Really think eval() should be avoided, if we actually want dynamic code in the translations a `new Function(...)` should probably be used instead and not let the function have access to global scope.

Reply:
Yes, eval-ing a function ís generally frowned upon. Hopefully this goes away when @uncleLem removes the function from the translations

Reply:
Replacing `eval` with `Function` is a good idea, I didn't think about it because I took these calendars directly from moments.js as they were and only changed the time part.
The differences in grammar wouldn't allow to have one approach for all, and I believe having all different variants in the calendar.js (like the moments.js stores it) would increase its size significantly and make it harder to maintain - all the calendars for all the languages existing in translations at the moment would add at least 350 lines. That's why I thought it would be better to keep language-specific part of it separate from the main logic in existing language-specific files.
But I don't insist on using translation files, I also think it may be confusing, but it looked right at the moment. Perhaps, a separate js file for storing these calendars would be a better solution? What do you think?

Reply:
Since I neither speak polish nor russian I need a little bit of help to understand what you are trying to do with the function.

I'm guessing the function is used for the fourth row in your example ("Wednesday") where it has some special grammar depending on the weekday and just presenting the name of the weekday isn't logical?

- Today and Tomorrow are already in the translation-files, would be better to not have duplicate translations for the same thing.
- The weather-module would have the same problem then since it displays "Today, Tomorrow, <weekday name>"

Reply:
@buxxi Yes, the function is required for some grammatical nuances. Sometimes it's the difference in prepositions for different weekdays, sometimes there are different grammatical cases that won't match simple `dddd` format, and also some day names are feminine, some are masculine, and some are neutral, changing the declension of related words. 
Simply displaying the name of a weekday like in English would be grammatically correct (for the languages I know), but in uk/pl/ru it would sound kinda synthetic and official, which is not exactly the aim of the relative time format from my understanding, it would look more like the absolute date format with weekdays. It seems to me that the English calendar in moment.js implementation also could have had "on" before the weekday, but moment.js devs implemented it differently from others in this regard.

> Today and Tomorrow are already in the translation-files, would be better to not have duplicate translations for the same thing.

You are totally right, it can be replaced (at least, for these languages) and I'll do it.

> The weather-module would have the same problem then since it displays "Today, Tomorrow, "

Is it? I believe it shows simply the weekdays, and I couldn't find any mentions of today/tomorrow. Also, my changes shouldn't affect other modules in any way.